spec/unit/govuk_index/presenters/indexable_content_presenter_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+
+RSpec.describe GovukIndex::IndexableContentPresenter do
+  it "extracts govspeak and text html from body fields" do
+    details = {
+      "body" => [
+        { "content_type" => "text/govspeak", "content" => "**hello**" },
+        { "content_type" => "text/html", "content" => "<strong>hello</strong>" }
+      ]
+    }
+
+    expect(indexable_content_presenter(details).indexable_content).to eq("hello")
+  end
+
+  it "extracts contact indexable content correctly" do
+    details = {
+      "some_other_key" => "value",
+      "title" => "Title",
+      "description" => "Description",
+      "contact_groups" => [
+        {
+          "slug" => "slug-1",
+          "title" => "Title 1"
+        },
+        {
+          "slug" => "slug-2",
+          "title" => "Title 2"
+        }
+      ]
+    }
+
+    expect(indexable_content_presenter(details, 'contact').indexable_content).to eq("Title\n\n\nDescription\n\n\nTitle 1\n\n\nTitle 2")
+  end
+
+  it "extracts content from details with parts" do
+    details = {
+      "parts" => [
+        {
+          "title" => "title 1",
+          "slug" => "title-1",
+          "body" => [
+            { "content_type" => "text/govspeak", "content" => "**hello**" },
+            { "content_type" => "text/html", "content" => "<strong>hello</strong>" }
+          ],
+        },
+        {
+          "title" => "title 2",
+          "slug" => "title-2",
+          "body" => [
+            { "content_type" => "text/govspeak", "content" => "**goodbye**" },
+            { "content_type" => "text/html", "content" => "<strong>goodbye</strong>" }
+          ],
+        }
+      ]
+    }
+
+    expect(indexable_content_presenter(details).indexable_content).to eq("title 1\n\nhello\n\ntitle 2\n\ngoodbye")
+  end
+
+  it "extracts additional indexable content keys when they have been specified" do
+    details = {
+      "external_related_links" => [],
+      "introductory_paragraph" => [
+        { "content_type" => "text/govspeak", "content" => "**introductory paragraph**" },
+        { "content_type" => "text/html", "content" => "<strong>introductory paragraph</strong>" }
+      ],
+      "more_information" => "more information",
+      "start_button_text" => "Start now",
+    }
+
+    expect(indexable_content_presenter(details, "transaction").indexable_content).to eq("introductory paragraph\n\nmore information")
+  end
+
+  def indexable_content_presenter(details, format = 'help_page')
```

Comment:
I would move this to a `subject` definition as that is more inline with standard rspec styling